frontend/src/views/consumer/event-listing/TicketInputSet.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { FlexBoxHorizontal } from './EventBooking';
+import { observer } from 'mobx-react';
+import { clamp } from '../../../utils';
+import Form, { InputField } from '../../../components/form';
+
+export default observer(
+  class TicketInputSet extends Component {
+    findTicketCatalogMaxSeats = id => {
+      const catalog = this.props.ticketCatalog.find(elem => elem.id == id);
+      if (catalog == undefined) {
+        return 0;
+      }
+      return catalog.availableSeatForThisType;
+    };
+    render() {
+      const { tickets, ticketCatalog } = this.props;
+      return (
+        <Form>
+          {tickets.map((ticket, index) => (
+            <FlexBoxHorizontal key={ticket.id || index}>
+              <InputField
+                placeholder="Lipputyppi"
+                value={ticket.label}
+                onChange={value => {
```

Comment:
This doesn't feel right to me.. the component is editing values coming through props?

Instead, the component should get the onChange callback as prop and the callback function should be defined in the comonent which is responsible of the ticket data.

Reply:
The data is observable so instead of making a method out of it you can modify it directly and the component (being an observer to it) can update accordingly.
This is more ergonomic than the approach of data ownership in the case of React's setState, where child components have no direct access to parent's state and must talk to it via a myriad of functions.
I think this is acceptable and easier to work within the context where:

* Direct CRUD updates on a model
* Direct child and not too far descendant

Which is the case for this component

Reply:
Ok. I'm not familiar with mobX and how observables work, so based on your explanation I trust you have made the right choice here. 👍